ofrak_core/ofrak/model/component_model.py

```diff
@@ -17,6 +19,62 @@ class ComponentConfig:
     """
 
 
+@dataclass
+class ComponentExternalTool:
+    """
+    An external tool or utility (like `zip` or `squashfs`) a component depends on
+    """
+
+    tool: str
+    install_hints: str = None  # e.g. version
+    install_check_arg: str = "--help"
+
+    async def run_tool(self, *args: str) -> str:
```

Comment:
I don't think `ComponentExternalTool` needs a `run_tool` method. 

First, it adds a level of indirection that makes the individual components less readable.
Second, it is valuable to let component implementations define how subprocesses are used. For example, see https://github.com/redballoonsecurity/ofrak/blob/master/ofrak_components/ofrak_components/cpio.py#L93, which differs from https://github.com/redballoonsecurity/ofrak/blob/master/ofrak_components/ofrak_components/tar.py#L88.

What I think does make sense is a `get_command` method, that returns `zip`, or `tar` command.

Reply:
I implemented some additions to the API that allow supporting such edge cases pretty trivially.

I get that using shell scripting is intuitive, however it's also probably less portable and more vulnerable. My latest examples show that Python & OFRAK allow us to do the same thing without too much more code.

Reply:
Removed the `run_tool`, we shouldn't needlessly herd contributors to a method arbitrarily.

I did leave my changes to those edge cases qualitatively intact, to remove the `shell=True` param and show that `subprocess` API does allow these very script-like usages, with hardly any extra code.